aspnetcore/blazor/includes/js-interop/circuit-disconnection.md

```diff
@@ -0,0 +1,46 @@
+*This section only applies to Blazor Server apps.*
+
+JavaScript (JS) interop calls can't be issued after a SignalR circuit is disconnected. Without a circuit during component disposal or at any other time that a circuit doesn't exist, the following method calls fail and log a message that the circuit is disconnected as a <xref:Microsoft.JSInterop.JSDisconnectedException>:
+
+* JS interop method calls
+  * <xref:Microsoft.JSInterop.IJSRuntime.InvokeAsync%2A?displayProperty=nameWithType>
+  * <xref:Microsoft.JSInterop.JSRuntimeExtensions.InvokeAsync%2A?displayProperty=nameWithType>
+  * <xref:Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync%2A?displayProperty=nameWithType>)
+* `Dispose`/`DisposeAsync` calls on any <xref:Microsoft.JSInterop.IJSObjectReference> or `Dispose` calls on any <xref:Microsoft.JSInterop.DotNetObjectReference>.
+
+In order to avoid logging <xref:Microsoft.JSInterop.JSDisconnectedException> or to log custom information, catch the exception in a [`try-catch`](/dotnet/csharp/language-reference/keywords/try-catch) statement.
+
+For the following component disposal example with an <xref:Microsoft.JSInterop.IJSObjectReference> object used to call JS from .NET:
+
+* The component implements <xref:System.IAsyncDisposable>.
+* `objInstance` is an <xref:Microsoft.JSInterop.IJSObjectReference>.
+* <xref:Microsoft.JSInterop.JSDisconnectedException> is caught and not logged.
+* Optionally, you can log custom information in the `catch` statement at whatever log level you prefer. The following example doesn't log custom information because it assumes the developer doesn't care about when or where circuits are disconnected during component disposal.
+
+```csharp
+async ValueTask IAsyncDisposable.DisposeAsync()
+{
+    try
+    {
+        if (objInstance is not null)
+        {
+            await objInstance.DisposeAsync();
+        }
+    }
+    catch (JSDisconnectedException)
+    {
+    }
+}
+```
+
+> [!NOTE]
+> The pattern for <xref:Microsoft.JSInterop.DotNetObjectReference> disposal is the same in scenarios where .NET is called from JS.
```

Comment:
I'm not sure what this means. Could we have an example for disposal "where .NET is called from JS"?

Reply:
It's the same disposal concept (trap `JSDisconnectedException`), but for a `DotNetObjectReference` in a `Dispose` method. Actually, I'm not sure if calling `Dispose` on an `DotNetObjectReference` will trigger a `JSDisconnectedException` if the circuit is dead. I'll try to run a test on that.

Reply:
> The JSDisconnectedException isn't thrown in the in the DotNetObjectReference case.
Oh, I think it is thrown unfortunatelly, at least in .NET 6

Reply:
I couldn't repro that here in a 6.0 test app. It didn't throw. You'd need to give me exact repro steps. Also, I recommend not commenting on this aspect further here if you are able to present a step-by-step repro of it because we don't work off of closed issues/PRs. It's best to open a new docs issue from the bottom of the topic.